### PR TITLE
remove collection and use typed_index_collection

### PR DESCRIPTION
**Pull request by datanel**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -95,6 +93,6 @@ name = "write_netex_france"
 path = "tests/write_netex_france.rs"
 required-features = ["proj"]
 
-[dev-dependencies]
+[dev_dependencies]
```
Comment (woshilapin) [inline Cargo.toml]: Hmm, official documentation mentioned with a dash `-`.

https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#development-dependencies

File: src/gtfs/write.rs [hunk 2]
```diff
@@ -537,7 +537,7 @@ mod tests {
     use pretty_assertions::assert_eq;
     use std::{collections::BTreeSet, fs::File, io::Read};
     use tempfile::tempdir;
-    use transit_model_collection::CollectionWithId;
+    use typed_index_collection::CollectionWithId;
```
Comment (woshilapin) [inline src/gtfs/write.rs]: This line can probably be removed? It's also at the top of the file.

File: tests/merge_ntfs.rs [hunk 3]
```diff
@@ -23,20 +23,22 @@ use transit_model::objects::{Comment, CommentLinks, StopPoint, VehicleJourney};
 use transit_model::test_utils::*;
 use transit_model::transfers;
 use transit_model::transfers::TransfersMode;
-use transit_model_collection::CollectionWithId;
-use transit_model_collection::Idx;
+use typed_index_collection::CollectionWithId;
+use typed_index_collection::Idx;
```
Comment (woshilapin) [inline tests/merge_ntfs.rs]: Maybe they can be merged?
```suggestion
use typed_index_collection::{CollectionWithId, Idx};
```

File: Cargo.toml (hunk 1 above)
Comment (datanel) [reply Cargo.toml]: I know, I didn't change that, weird

File: tests/merge_ntfs.rs (hunk 3 above)
Comment (datanel) [reply tests/merge_ntfs.rs]: maybe :)

**Pull request merged**